Feat/#tc 116 create template selection on family tree

Teodor25 opened the pull request:
### This pull request closes:

✔️ Closes Jira Issue: [TC-116](https://treecreate.atlassian.net/browse/TC-116)

### Which service(s) does this issue affect:

- [ ] 🛰️ API
- [x] 🛒 webstore
- [ ] 🎓 admin-page

### Types of changes

- [ ] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [x] 🧱 New feature (a non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🤖 CI/CD (a change to the CI/CD pipeline)
- [ ] ⚗️ Refactoring (an update to some already existing code)
- [ ] 💄 Style (Markup, formatting, CSS)

### Documentation Updates

- [ ] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [x] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [x] Tested on Safari
- [x] Tested on Chrome
- [x] Tested on Mozilla Firefox

### What changes have been done within this issue?

a new function to start a family tree design from a teamplate. 
First time you visit product page the template modal will pop up. it is stored in localstorage if its your first visit so it doesnt show second time you visit the page. 
There is a new button in options menu where you can open the template modal. 

### How should this be manually tested?

go to product page and select a template. 

### Screenshots or example usage

<img width="765" alt="Screenshot 2022-03-04 at 23 56 13" src="https://user-images.githubusercontent.com/56150163/156853219-34a07d2a-77e2-4187-afab-1e6eb8644253.png">

Teodor25 committed:
feat: 🎸 Initiate template function and add button to options

Teodor25 committed:
feat: 🎸 Initiate modal for template

Teodor25 committed:
feat: 🎸 Add famtree miniature to modal display

Teodor25 committed:
feat: 🎸 Create and add templates

Teodor25 committed:
feat: 🎸 Create TemplateEnum for better use of templates

Teodor25 committed:
feat: 🎸 Apply template to design funciton

Teodor25 committed:
feat: 🎸 Add localstorage variable to show template

Teodor25 committed:
feat: 🎸 Add start from scratch function to template

Teodor25 committed:
test: 💍 Fix e2e tests for product page

Teodor25 committed:
test: 💍 Add e2e tests to template functionality

Kwandes commented on libs/interfaces/src/lib/design/design.interface.ts:
```diff
@@ -50,3 +51,9 @@ export interface IFamilyTreeBanner {
   style: 'first';
 }
+
+export interface Template {
```
interfaces should reflect their existance in the variable name
```suggestion
export interface ITemplate {
```
This change has to be applied to all instanced of Template

You should also change the name to something more descriptive like `IFamilyTreeTemplate`

Kwandes commented on apps/webstore/src/app/pages/product/product.component.ts:
```diff
@@ -104,6 +113,14 @@ export class ProductComponent implements OnInit {
+    this.modalService.open(FamilyTreeTemplateModalComponent);
+  }
+
+  selectTemplate(name: string): void {
```
This method is not called by anything. Why does it exist?

Kwandes commented on apps/webstore/src/app/shared/components/modals/family-tree-template-modal/family-tree-template-modal.component.ts:
```diff
@@ -0,0 +1,23 @@
+
+  constructor(public activeModal: NgbActiveModal, private localStorageService: LocalStorageService) {}
+
+  applyTemplate(name: string) {
```
Missing jsdoc and return type (void)

Kwandes commented on apps/webstore/src/app/shared/components/modals/family-tree-template-modal/family-tree-template-modal.component.ts:
```diff
@@ -0,0 +1,23 @@
+  applyTemplate(name: string) {
+    const selectedTemplate: Template = templates.find((template) => template.name === name);
+    this.localStorageService.setItem<IFamilyTree>(LocalStorageVars.designFamilyTree, selectedTemplate.designProperties);
+    location.reload();
```
You need to remove the designId from the URL parameters. This currently doesn't work if the user is viewing a design via the ID (when logged in)

Teodor25 committed:
refactor: 💡 Optimize naminconvention

Teodor25 committed:
refactor: 💡 Remove redundant code

Teodor25 committed:
refactor: 💡 Add JSDocs

Teodor25 committed:
refactor: 💡 resolve commit messages

Teodor25 committed:
fix: 🐛 Fix background not loading

Teodor25 committed:
feat: 🎸 Add option to see more templates

Teodor25 committed:
test: 💍 Fix intro modal failing e2e tests

Teodor25 committed:
test: 💍 Fix e2e tests for template modal

Teodor25 committed:
test: 💍 Add test for seeing more templates

Teodor25 committed:
feat: 🎸 Add new templates

Teodor25 committed:
feat: 🎸 Add localisation ids

Teodor25 committed:
test: 💍 Fix e2e test for template modal

Teodor25 committed:
feat: 🎸 Add localization text and formatting

Teodor25 committed:
fix: 🐛 Fix bug with selecting a template

Teodor25 committed:
feat: 🎸 Change Family tree box sizing limits and default

Teodor25 committed:
feat: 🎸 Add new browser icon

Teodor25 committed:
refactor: 💡 Remove unnecessary images and duplicate icons

Teodor25 committed:
Merge branch 'development' into feat/#TC-116-create-template-selection-on-family-tree

Teodor25 committed:
refactor: 💡 Remove signup section in home page

Since we have a newsletter popup modal and a signup in the footer, we
dont need one in the home page.

Teodor25 committed:
test: 💍 Fix failing e2e

Teodor25 committed:
style: 💄 Apply formatting

Teodor25 committed:
test: 💍 fix e2e tests for template modal

Teodor25 committed:
test: 💍 Fix basket e2e tests

Teodor25 committed:
test: 💍 Fix e2e tests for product page

Teodor25 committed:
test: 💍 Fix ci tests for template modal

Teodor25 committed:
feat: 🎸 Add new template to template modal

Teodor25 committed:
style: 💄 Apply formatting

Teodor25 committed:
test: 💍 Fix e2e test failing

Teodor25 committed:
Merge remote-tracking branch 'origin/development' into feat/#TC-116-create-template-selection-on-family-tree

Kwandes commented on apps/webstore/src/app/shared/components/modals/family-tree-template-modal/family-tree-template-modal.component.ts:
```diff
@@ -0,0 +1,46 @@
+  /**
+   * Sets the local storage design to the selected template
+   *
+   * @param name
```
You forgot to describe the `name` param and a return type

Kwandes commented on apps/webstore/src/app/shared/components/modals/family-tree-template-modal/family-tree-template-modal.component.ts:
```diff
@@ -0,0 +1,46 @@
+    location.reload();
+  }
+
+  returnTemplateList() {
```
You forgot javadoc and return type

Teodor25 committed:
refactor: 💡 Add better comment based of github

Teodor25 committed:
refactor: 💡 rename templates

Teodor25 committed:
feat: 🎸 Add templates as images

Teodor25 committed:
feat: 🎸 remove minitree from template and set img

Teodor25 committed:
feat: 🎸 Add alt description to img in template modal

Teodor25 committed:
style: 💄 Apply formatting

Teodor25 committed:
refactor: 💡 rename spec file

Teodor25 committed:
style: 💄 Add JDocs

Teodor25 committed:
style: 💄 Apply formatting
